provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/exceptions/ProvisioningDeviceClientExceptionManager.java

```diff
@@ -18,38 +18,30 @@ public static void verifyHttpResponse(HttpResponse httpResponse) throws Provisio
     {
         int responseStatus = httpResponse.getStatus();
 
-        String errorMessage = ErrorMessageParser.bestErrorMessage(new String(httpResponse.getErrorReason(), StandardCharsets.UTF_8));
+        byte[] errorReason = httpResponse.getErrorReason();
+
+        String errorMessage = httpResponse.getStatus() + " : " + new String(errorReason, StandardCharsets.UTF_8);
 
         switch (responseStatus)
         {
             case 400:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 401:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 403:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 404:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 412:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 429:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 500:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 502:
-                throw new ProvisioningDeviceHubException(errorMessage);
             case 503:
```

Comment:
This was a strange switch case where almost all cases did the exact same thing. Just using a fall through to simplify it a bit